### DISCUSSION TITLE
Specifics of the yi-hack-v5 firmware and flashing

### DISCUSSION THREAD

**Discussion by alienatedsec** [alienatedsec/yi-hack-v5#267] [Show and tell]
`[Updated] 2024/03/01`

There are certain complexities with the project and what it was based on; `yi-hack-v3`, and `yi-hack-v4`.
The above versions are optimal for what was required at the time and reside within the device's `rootfs` and `home` partitions.

This has evolved with `yi-hack-v5` - the `rootfs` is too small to hold all binaries, and `home` is not flashing when packed with more files. As an outcome, the `yi-hack-v5` folder is located on the SD Card; however, the `yi-hack-v5` needs the initial baseline hack to `rootfs` and `home` partitions too.

- The latest baseline version is [`0.4.1`](https://github.com/alienatedsec/yi-hack-v5/releases/tag/0.4.1). 

**The `yi-hack-v5` comes with several challenges:**
- To load the baseline hack into `rootfs` and `home` partitions, you need to prepare the SD Card for it. The below text was copied from the project site.

You can get a microSD card, preferably of capacity 16gb or less and format it by selecting File System as FAT32.

**_IMPORTANT: The microSD card must be formatted in FAT32. exFAT formatted microSD cards will not work._**
**I have not formatted any of my 32GB cards to load the firmware. Just copy files directly and it should work.**

<summary> How to format microSD cards > e.g., 32GB as FAT32 in Windows 10</summary><p>

For microSD cards larger than 32 GB, Windows 10 only gives you the option to format as NTFS or exFAT. You can create a small partition (e.g 4 GB) on a large microSD card (e.g. 64 GB) to get the FAT32 formatting option.

* insert microSD card into PC card reader
* open Disk Management (e.g. <kbd>Win</kbd>+<kbd>x</kbd>, <kbd>k</kbd>)
  * Disk Management: delete all partitions on the microSD card
    * right click each partition > "Delete Volume..."
    * repeat until there are no partitions on the card
  * Disk Management: create a new FAT32 partition
    * right click on "Unallocated" > "New Simple Volume..."
    * Welcome to the New Simple Volume Wizard: click "Next"
    * Specify Volume Size: 4096 > "Next"
    * Assign Drive Letter or Path: (Any) > "Next"
    * Format Partition: Format this volume with the following settings:
      * File system: FAT32
      * Allocation unit size: Default
      * Volume label: Something
      * Perform a quick format: &#9745;

You should now have a FAT32 partition on your microSD card that will allow the camera to load the firmware files to update to `yi-hack-v5`.

### Example: 4 GB FAT32 partition on 64 GB microSD card
![image](https://github.com/alienatedsec/yi-hack-v5/assets/73167064/2c4b7b41-85df-4abb-a694-3f63091b848f)

Alternative way:
* open cmd with admin permissions
* run diskpart
* type "list disk"
* find your sd card (for example Disk 7)
* type "select disk 7"
* if it has one partition - type "select partition 1". If more - delete all the partitions and then create one
* type "format FS=FAT32 QUICK"
* done. 32GB partition in FAT32.

</p>

**IMPORTANT STEPS**
- Once you copy your firmware files and the `yi-hack-v5` folder, it will flash with the initial baseload files.
- Once flashed, there will likely be no video recording - this is because the SD card is formatted as FAT32, and this will be seen as an error by the camera's firmware.
- To enable video recording, you need to format the SD card with the `YI App`.
- It will remove everything and once again you have to copy the `yi-hack-v5` folder for the hack to work again. You can simply copy the folder over FTP as this will be enabled if the `yi-hack-v5` folder does not exist during a bootup process.
- Reboot the camera after copying the folder.
- I have made some pre-checks if the baseline firmware is loaded correctly - if you find `fat32error.txt` in `/tmp/sd` then your version file does not exist and your firmware was flashed incorrectly.
- run `./script/launch.sh` from the console to check the versions installed
![image](https://github.com/alienatedsec/yi-hack-v5/assets/73167064/c3c3ce9e-cfa4-4e6d-a466-a0202195f410)
- I plan on keeping the version on cam as `0.4.1` and this will unlikely change, unless we find the `Busybox 1.36.1` buggy and need updating again.

**Comment by lbls55** [comment]
Hi friend, first thanks to your incredible work. 
I've installed you last pre release version (0.4.1g) on my Yi Dome 1080p, it's working, but I have 2 questions: 
In which folder do I have to copy (by FTP as you said) the Yi-hack v5 folder again after formatting the SD card with the Yi app?
And my second question is: how to set ptz presets, when i move the cam using arrows to a specified position, I give a name to the preset and after I click on "save preset " button, nothing happens.
Thanks for helping.
Laurent

**Reply by alienatedsec** [reply]
Please copy the yi-hack-v5 folder onto /tmp/sd/ 

In regards to presets, I would need to double-check on my yi_dome but it used to save in the YI app and restore via a hack WebGUI

**Reply by lbls55** [reply]
ok, thanks, I only need yi-hack-v5 folder, nothing else? no rootfs_h20 and no home_h20?

**Reply by alienatedsec** [reply]
Correct. Just to be aware of the ongoing work which requires updating the baseline until the official latest release

**Comment by othyro** [comment]
I was looking to update one of my cameras today (Hardware ID: 47US), because it's running hack (SD) firmware version 0.4.0 with  baseline firmware version 0.3.8. In the web UI, this version doesn't show as available (blank). IIUC, I need to do an install from scratch following the above steps **and** use the Yi app. Does the desktop version of the latter work for this? I have no desire to use their mobile app.

This project's versioning is a bit misleading. The .z in x.y.z is a _revision_ increment. This update sounds like a whole major version based on the procedure I'm reading. Is this going to be the same process **every time** you update the baseline version? If so, I'm not seeing the benefit of upgrading, because this seems painful.

**Comment by SebiC98** [comment]
Hello, guys.

@othyro @alienatedsec @lbls55 

I managed to copy via ftp the folder, but i can not run the install via ssh. I get permission denied. How can I run the install, I still can not connect to the web config.

**Reply by SebiC98** [reply]
I managed to create a .sh script that runs when the camera reboots in order to bypass the problem with ssh. If anyone needs help contact me.

I just added a .sh file that automatically runs ./script/launch.sh when the camera reboots.

**Reply by muava12** [reply]
i have new sd card, so what folder i have to create and copy for yi-home?

**Reply by gppapas-max** [reply]
I have copy the folder to /tmp/sd but I still can not open through browser the camera. What I have to do now?

**Comment by RussellRego** [comment]
I have a Keyke 1080p outdoor cam and I am trying to get RTSP feed.  I followed grausof keyke-hack and am able to Telnet and FTP into the device.  No other ports seem to be open, but the camera streams to the Chinese Company's app and now I cannot even replay my own recordings on the SD Card without constand ads and harassment to buy cloud recording.

I followed posts to make a donation on Discord when I saw this yi-hack-5.  I am wondering if this will work on my cam.  Could anyone direct me to how to find out what Yi firmware I have and directions from there?  

If this is not the correct place to request this kind of assistance, please direct me.  TIA

**Comment by lovelyelfpop** [comment]
Is sd card still required even if the firmware is flashed? (I don't need video recording)

**Reply by LinoBarreca** [reply]
Yeah I would like to know this as well...from the original comment it seems like the folder has some code to **copy** additional files to the home (this means that once the home folder has been "filled" with the proper files, you shouldn't need the SD anymore)...but I might be wrong here and a confirmation would be nice.

**Comment by LinoBarreca** [comment]
OK. I did way more than waiting for confirmation.
I branched this and now I can run without SD card, with a new UI, improved binaries and it's just the tip of the optimizations possible.
Ladies and gentleman let me introduce you [yi-hack-v6](https://github.com/LinoBarreca/yi-hack-v6)
<img width="1050" height="703" alt="Screenshot 2026-07-21 at 11-04-09 CamDevelopment - Live view" src="https://github.com/user-attachments/assets/a9b62f53-1eb2-4ac9-a5c9-480875f6e0d1" />
There are no releases yet because I am still not sure what files I am allowed to distribute but you can build the source code yourself and/or you can contact me if you want to test or report bugs.
